HotSpot/View Controllers /LocationManagerSingleton.m

```diff
@@ -0,0 +1,54 @@
+//
+//  LocationManager.m
+//  HotSpot
+//
+//  Created by aodemuyi on 8/7/19.
+//  Copyright © 2019 aodemuyi. All rights reserved.
+//
+
+#import "LocationManagerSingleton.h"
+
+@implementation LocationManagerSingleton
+
+@synthesize locationManager;
+
++ (id)sharedLocationManager {
+    static LocationManagerSingleton *sharedLocationManager = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        sharedLocationManager = [[self alloc] init];
+    });
+    return sharedLocationManager;
+}
+- (id)init {
+    self = [super init];
+    
+    if(self) {
+        self.locationManager = [[CLLocationManager alloc] init];
+        self.locationManager.delegate = self;
+        [self.locationManager startUpdatingLocation];
+        [self.locationManager setDistanceFilter:kCLDistanceFilterNone];
+        [self.locationManager setHeadingFilter:kCLHeadingFilterNone];
+        [self.locationManager requestWhenInUseAuthorization];
+        //do any more customization to your location manager
+    }
+    
+    return self;
+}
+
++ (LocationManagerSingleton*)sharedSingleton {
```

Comment:
lol what's the difference between this method and "sharedLocationManager"? they do the same thing!

Reply:
sorry didn't realize I wrote it twice